network-formula/network/wicked/netconfig.sls

```diff
@@ -0,0 +1,52 @@
+{#-
+Salt state file for managing the general network configuration
+Copyright (C) 2023 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+-#}
+
+{%- from 'network/wicked/map.jinja' import base, config -%}
+
+network_wicked_config_header:
+  file.prepend:
+    - name: {{ base }}/config
+    - text: {{ pillar.get('managed_by_salt_formula', '# Managed by the network formula') | yaml_encode }}
+
+{%- if config %}
+network_wicked_config:
+  file.keyvalue:
+    - name: {{ base }}/config
+    - key_values:
+      {%- for key, value in config.items() %}
+      {%- if value is sameas true %}
+      {%- set value = 'yes' %}
+      {%- elif value is sameas false %}
+      {%- set value = 'no' %}
+      {%- elif value is iterable and value is not string %}
+      {%- set value = ' '.join(value) -%}
+      {%- else %}
+      {%- set value = value | lower %}
```

Comment:
Just wondering - should `' '.join(value)` also get a `| lower`?

Reply:
I was thinking about it and deemed it not necessary as it's probably mostly used for DNS server IP addresses. Though it might be good for consistency.